src/roiextractors/extractors/tiffimagingextractors/brukertiffimagingextractor.py

```diff
@@ -0,0 +1,123 @@
+        return tree.getroot()
+
+    def _get_xml_metadata(self):
+        root = self._get_xml_root()
+        xml_metadata = dict()
+        xml_metadata.update(**root.attrib)
+        for child in root.findall(".//PVStateValue"):
+            metadata_root_key = child.attrib["key"]
+            if "value" in child.attrib:
+                if metadata_root_key in xml_metadata:
+                    continue
+                xml_metadata[metadata_root_key] = child.attrib["value"]
+            else:
+                xml_metadata[metadata_root_key] = []
+                for indexed_value in child:
+                    if "value" not in indexed_value.attrib:
+                        for subindexed_value in indexed_value:
+                            if "description" in subindexed_value.attrib:
+                                xml_metadata[metadata_root_key].append(
+                                    {subindexed_value.attrib["description"]: subindexed_value.attrib["value"]}
+                                )
+                            else:
+                                xml_metadata[child.attrib["key"]].append(
+                                    {indexed_value.attrib["index"]: subindexed_value.attrib["value"]}
+                                )
+                    if "description" in indexed_value.attrib:
+                        xml_metadata[child.attrib["key"]].append(
+                            {indexed_value.attrib["description"]: indexed_value.attrib["value"]}
+                        )
+                    elif "value" in indexed_value.attrib:
+                        xml_metadata[child.attrib["key"]].append(
+                            {indexed_value.attrib["index"]: indexed_value.attrib["value"]}
+                        )
+        return xml_metadata
+
+    def _get_files_names(self):
+        return [file.attrib["filename"] for file in self._get_xml_root().findall(".//File")]
+
+    def get_image_size(self) -> Tuple[int, int]:
+        return self._height, self._width
+
+    def get_num_frames(self) -> int:
+        return len(self._file_paths)
+
+    def get_sampling_frequency(self) -> float:
+        frame_period = float(self.xml_metadata["framePeriod"])
+        return 1 / frame_period
+
+    def get_channel_names(self) -> list:
+        channel_names = [file.attrib["channelName"] for file in self._get_xml_root().findall(".//File")]
+        unique_channel_names = list(set(channel_names))
+        return unique_channel_names
+
+    def get_num_channels(self) -> int:
+        channel_names = self.get_channel_names()
+        return len(channel_names)
+
+    def get_dtype(self) -> DtypeType:
+        return self._dtype
+
+    def _frames_iterator(
+        self,
+        start_frame: Optional[int] = None,
+        end_frame: Optional[int] = None,
+    ):
+        for file in self._file_paths[start_frame:end_frame]:
+            yield _get_tiff_reader().memmap(self.folder_path / file, mode="r", _multifile=False)
```

Comment:
```suggestion
    ):
        tiffile = get_package(package_name="tifffile", installation_instructions="pip install tifffile")

        for file in self._file_paths[start_frame:end_frame]:
            yield tiffile.memmap(self.folder_path / file, mode="r", _multifile=False)
```

reads a little awkward to re-import on each iteration; I know it technically won't affect performance hardly at all but this should work the same way

Reply:
You're right, thanks!